pkg/provider/azure.go

```diff
@@ -0,0 +1,188 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/reference"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	expcapz "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	expcapi "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure"
+	"github.com/giantswarm/sonobuoy-plugin/v5/pkg/azure/credentials"
+)
+
+const (
+	defaultVMSize = "Standard_D4s_v3"
+)
+
+type AzureProviderSupport struct {
+	azureClient *azure.Client
+}
+
+func NewAzureProviderSupport(ctx context.Context, client ctrl.Client, cluster *capi.Cluster) (Support, error) {
+	sp, err := credentials.ForCluster(ctx, client, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	azureClient, err := azure.NewClient(azure.ClientConfig{ServicePrincipal: sp})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	p := &AzureProviderSupport{
+		azureClient: azureClient,
+	}
+
+	return p, nil
+}
+
+func (p *AzureProviderSupport) CreateNodePool(ctx context.Context, client ctrl.Client, cluster *capi.Cluster, azs []string) (*expcapi.MachinePool, error) {
+	azureMP, err := p.createAzureMachinePool(ctx, client, cluster, azs)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	mp, err := p.createMachinePool(ctx, client, cluster, azureMP, azs)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return mp, nil
+}
+
+func (p *AzureProviderSupport) GetProviderAZs() []string {
+	return []string{"1", "2", "3"}
```

Comment:
This will never work in germany west central. Nothing major, but just to be aware of this should we ever decide to run tests in GWC.

Reply:
Yeah, I was thinking about this. This was kind of "works for now". By changing signature to `GetProviderAZs(region string) ([]string, error)` we could actually implement proper query from provider, but I'm not sure if it's worth it for now.